src/HexViewer.ts

```diff
@@ -177,12 +179,12 @@ export class HexViewer {
                 e.key === "ArrowRight" ? 1 : e.key === "ArrowLeft" ? -1 : null;
 
             if (selDiff === null) return;
-
-            var newSel = this.selectionStart + selDiff;
-            if (newSel < 0) newSel = 0;
-            else if (newSel >= this.dataProvider.length) newSel = this.dataProvider.length - 1;
```

Comment:
These indispensable checks are removed without any compensation. As a result, when you try to seek before the beginning using the keyboard keys (place the cursor somewhere on the first line of the hex dump and then press the **Up Key `↑`**), some _weird_ stuff happens and soon the hex dump becomes completely unusable. Trying to interact with the hex dump using the keyboard or the mouse doesn't do anything. AFAIK the only way how to get things back to normal is to reload the page. So please at least bring them back so that this new implementation isn't broken more that the existing one.

To be fair, this adjustment apparently solves the problem with the keyboard, but a similar problem persists using the mouse (clicking on some invisible "byte" below the hex dump). Here's the capture:

![kaitai-struct-webide-mouse-click-outside](https://user-images.githubusercontent.com/47499687/88463552-e2565d00-ceb3-11ea-82b3-a10bdcb475bf.gif)

It would be nice if you could solve it as well, preferably by moving this check that you removed here to some place common for both keyboard and mouse interaction, for example the `setSelection` method.  There are already some checks, but they apparently aren't enough.